clang/lib/CIR/Dialect/Transforms/TargetLowering/LowerCall.cpp

```diff
@@ -44,8 +50,150 @@ arrangeFreeFunctionLikeCall(LowerTypes &LT, LowerModule &LM,
                                     fnType.getInputs(), required);
 }
 
+/// Adds the formal parameters in FPT to the given prefix. If any parameter in
+/// FPT has pass_object_size attrs, then we'll add parameters for those, too.
+static void appendParameterTypes(SmallVectorImpl<Type> &prefix, FuncType fnTy) {
+  // Fast path: don't touch param info if we don't need to.
+  if (/*!fnTy->hasExtParameterInfos()=*/true) {
```

Comment:
If this is triggering a compiler warning just make sure you make it silent before landing.